src/main/java/com/imsweb/datagenerator/utils/DistributionUtils.java

```diff
@@ -179,4 +182,20 @@ public static String getState() {
             _DIST_STATE = Distribution.of(_STATES);
         return _DIST_STATE.getValue();
     }
+
+
+    public static int getAgeGroup(String site) {
+        if ((site == null) || (Thread.currentThread().getContextClassLoader().getResource("frequencies/sites/" + site.toUpperCase() + ".csv") == null))
+            return -1;
+
+        Distribution<String> distribution = _DIST_SITE_AGE_GROUPS.get(site.toUpperCase());
+        if (distribution == null) {
+            distribution = Distribution.of(Thread.currentThread().getContextClassLoader().getResource("frequencies/sites/" + site.toUpperCase() + ".csv"));
+            _DIST_SITE_AGE_GROUPS.put(site.toUpperCase(), distribution);
+        }
+        String ageGroup = distribution.getValue().trim();
```

Comment:
This local variable is not needed, you can just add the logic to the return statement; in this case that won't make the call any less readable and it will be a bit smaller (and technically a bit more efficient, although that part really doesn't matter). Also, why do you need to trim? Didn't you create this CSV file yourself? Can't you pre-trim the values so you don't have to do it each time a new frequency is requested?

Reply:
Fixed.